controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -559,6 +562,44 @@ func (r *BareMetalHostReconciler) actionMatchProfile(prov provisioner.Provisione
 	return actionComplete{}
 }
 
+func (r *BareMetalHostReconciler) actionPreparing(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	info.log.Info("preparing")
+
+	// Save provisioning settings.
+	provisioningSettings := info.host.Status.Provisioning.DeepCopy()
+	dirty, err := saveHostProvisioningSettings(info.host)
```

Comment:
@Hellcatlk, just to clarify, IIUC @zaneb 's [comment](https://docs.google.com/spreadsheets/d/1cyKN434j5FkjuAvSDsBq4IBdGlZNGa6jDAda7bre-rU/edit#gid=188658650) then `saveHostProvisioningSettings` will contains the necessary logic to trigger the manual cleaning in the Prepare. Right now in this method just the `Provisioning.RootDeviceHints` are checked. If it is expected to be enriched in future for the related RAID section, then probably I'd find more clear to add comment for that within such method

Reply:
That's correct. But note that this function already existed, and this is the right thing to do regardless of whether RAID is added.

Reply:
Right, I was thinking to something more like the `// TODO:` in the `buildManualCleaningSteps`